src/FormMigration/FormMetaDecorator.php

```diff
@@ -684,4 +683,29 @@ private function isDefaultFund(int $fundId): bool
 
         return $fund->id === $fundId;
     }
+
+    /**
+     * @unlreased
+     */
+    public function getGiftAidSettings(): array
+    {
+        $giftAidStatus = $this->getMeta('give_gift_aid_enable_disable');
+
+        if (empty($giftAidStatus) || ($giftAidStatus !== 'enabled' && $giftAidStatus !== 'global')) {
+            return [];
+        }
+
+        return [
+            'useGlobalSettings' => $giftAidStatus === 'global',
```

Comment:
I would caution against bundling like this in the `FormMetaDecorator`. These methods are intended to be human readable abstractions of the technical implementation and to decouple the shape of the data from how it is stored and used.

A few things to consider as a side effect of bundling the settings:
- Couples the block attributes shape with the database storage implementation
- Creates an un-typed return array which is prone to creating bugs when refactoring
- Returns different data depending on the value of the Gift Aid status

Reply:
I know it may seem like extra, unnecessary work to create the many methods, but it has advantages in future maintenance (which is always harder than the initial write).

Reply:
Refactored on https://github.com/impress-org/givewp/pull/7186/commits/479836b7c882cbdff8709affdc044d4d9dd485a3